common/shared.c

```diff
@@ -1837,7 +1837,13 @@ ssize_t ft_rx(struct fid_ep *ep, size_t size)
 static inline int ft_tag_is_valid(struct fid_cq * cq, struct fi_cq_err_entry *comp,
 				  uint64_t tag)
 {
-	if ((hints->caps & FI_TAGGED) && (cq == rxcq) && (comp->tag != tag)) {
+	if (opts.options & FT_OPT_BW) {
+		if ((hints->caps & FI_TAGGED) && (cq == rxcq) &&
+			((comp->tag >= tag + opts.window_size / 2) || (comp->tag + opts.window_size / 2 <= tag))) {
```

Comment:
Please limit line lengths to roughly 80 characters.  Not everyone uses a wide screen monitor to read/update code. :)